crates/metadata-store/src/omnipaxos/store.rs

```diff
@@ -0,0 +1,1232 @@
+            )) {
+                networking.register_address(*node_id, node_config.address.clone());
+            }
+        }
+
+        // todo remove additional indirection from Arc
+        connection_manager.store(Some(Arc::new(new_connection_manager)));
+
+        let omni_paxos = Self::create_omni_paxos(
+            own_member_id.node_id,
+            cluster_config.clone(),
+            rocksdb_storage,
+        );
+
+        let is_leader = omni_paxos
+            .get_current_leader()
+            .is_some_and(|(node_id, _)| node_id == own_member_id.node_id);
+
+        Active {
+            omni_paxos: Some(omni_paxos),
+            cluster_config,
+            members,
+            connection_manager,
+            networking,
+            msg_rx: router_rx,
+            own_member_id,
+            is_leader,
+            last_applied_index: 0,
+            kv_storage: KvMemoryStorage::new(metadata_writer.clone()),
+            request_rx,
+            join_cluster_rx,
+            pending_join_requests: HashMap::default(),
+            metadata_writer,
+        }
+    }
+
+    fn create_omni_paxos(
+        own_node_id: NodeId,
+        cluster_config: ClusterConfig,
+        rocksdb_storage: RocksDbStorage<Request>,
+    ) -> OmniPaxos {
+        let server_config = ServerConfig {
+            pid: own_node_id,
+            // todo make configurable
+            election_tick_timeout: 5,
+            resend_message_tick_timeout: 20,
+            ..ServerConfig::default()
+        };
+
+        let op_config = OmniPaxosConfig {
+            server_config,
+            cluster_config,
+        };
+
+        op_config
+            .build(rocksdb_storage)
+            .expect("omni paxos configuration is valid")
+    }
+
+    #[instrument(level = "info", skip_all, fields(member_id = %self.own_member_id))]
+    pub(crate) async fn run(mut self) -> anyhow::Result<Passive> {
+        debug!("Run as active metadata store node");
+
+        let mut tick_interval = time::interval(Duration::from_millis(100));
+        tick_interval.set_missed_tick_behavior(MissedTickBehavior::Burst);
+
+        let mut nodes_config_watch =
+            Metadata::with_current(|m| m.watch(MetadataKind::NodesConfiguration));
+        nodes_config_watch.mark_changed();
+
+        loop {
+            tokio::select! {
+                Some(request) = self.request_rx.recv() => {
+                    self.handle_request(request);
+                },
+                Some(request) = self.join_cluster_rx.recv() => {
+                    self.handle_join_request(request);
+                }
+                Some(msg) = self.msg_rx.recv() => {
+                    self.handle_omni_paxos_message(msg);
+                },
+                Ok(()) = nodes_config_watch.changed() => {
+                    self.update_node_addresses(&Metadata::with_current(|m| m.nodes_config_ref()));
+                },
```

Comment:
This is to support changing node/peer addresses that are communicated via the `NodesConfiguration`.